includes/Core/Permissions/Permissions.php

```diff
@@ -583,4 +595,19 @@ public static function get_dashboard_sharing_capabilities() {
 			self::DELEGATE_MODULE_SHARING_MANAGEMENT,
 		);
 	}
+
+	/**
+	 * Gets all the meta capabilities specifically added for dashboard sharing.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of meta capabilities specific to dashboard sharing.
+	 */
+	public static function get_dashboard_sharing_meta_capabilities() {
+		return array(
+			self::READ_SHARED_MODULE_DATA,
+			self::MANAGE_MODULE_SHARING_OPTIONS,
+			self::DELEGATE_MODULE_SHARING_MANAGEMENT,
+		);
+	}
```

Comment:
This function is called unconditionally, regardless of `Feature_Flags::enabled( 'dashboardSharing' )`, and returns the array unconditionally as well.

Would:

```suggestion
	public static function get_dashboard_sharing_meta_capabilities() {
		if ( ! Feature_Flags::enabled( 'dashboardSharing' ) ) {
			return array();
		}

		return array(
			self::READ_SHARED_MODULE_DATA,
			self::MANAGE_MODULE_SHARING_OPTIONS,
			self::DELEGATE_MODULE_SHARING_MANAGEMENT,
		);
	}
```

be more appropriate, so capabilities related to dashboard sharing are only ever returned when the feature flag is enabled?

If we do this, a test for that behaviour should be added to `PermissionsTest` as well.

Reply:
Thanks @tofumatt, makes sense 👍

Reply:
@tofumatt Good find - the caller of this method (`check_all_for_current_user`) isn't wrapping the logic within the Dashboard Sharing feature flag. I suggest we do that instead of returning an empty `array()`. This is because `get_dashboard_sharing_meta_capabilities()` is very similar to `get_dashboard_sharing_capabilities()` and the latter is being used almost as a 'constant' which needs to return the "dashboard sharing capabilities" even when the FF is disabled. In fact, it is used to specifically disallow a capability within `map_meta_cap()` when the FF is disabled. This will keep both of these methods consistent.

Reply:
Thanks @jimmymadon, I have addressed your comments 👍 
I have extracted `get_dashboard_sharing_meta_permissions` to a new function that checks the FF and returns early.

Reply:
Right on—that makes sense, thanks for the explanation @jimmymadon.

Thanks @hussain-t for making the change in-line with what @jimmymadon suggested, this looks good now 😄